yaks/lib/yaks/mapper/attribute.rb

```diff
@@ -2,22 +2,21 @@ module Yaks
   class Mapper
     class Attribute
       extend Forwardable, Util
-      include Util
-      include Attribs.new(:name, :block, options: {}.freeze)
+      include Attribs.new(:name, :block, if: true), Util
 
-      def self.create(*args, &block)
-        args, options = extract_options(args)
-        new(name: args.first, options: options, block: block)
+      def self.create(name, options = {}, &block)
+        new(options.merge({name: name, block: block}))
       end
 
       def add_to_resource(resource, mapper, _context)
-        return unless mapper.expand_value(options.fetch(:if, true))
```

Comment:
If i remember correctly mapper.expand_value could not access the `object` in the Proc